src/home/components/SearchBar.js

```diff
@@ -0,0 +1,40 @@
+import React, { Component } from 'react';
+import './SearchBar.css';
+
+class SearchBar extends Component {
+  state = {
+    query: ''
+  };
+
+  // función creada para ejecutar la consulta cuando se haga click en el botón del formulario o se pulse Intro
+  onFormSubmit = (event) => {
+    event.preventDefault();
+    this.props.onSubmit(this.state.query);
+  };
+  // función creada para transformar el formulario en un componente controlado
+  onInputChange = (event) => {
+    this.setState({ query: event.target.value });
+  };
+
```

Comment:
```
onInputChange = (event) => {
    this.setState({ query: event.target.value });
  };
```
Estamos en ES6. onFormSubmit no es un atributo que encapsula un array function sino un método de la clase. OJO CON ESTO QUE NO ES LO MISMO. Esto puede traer muchos errores. Cámbialo a como te indico. Si no comprendes cúal es la diferencia y los problemas que puede traer lo hablamos en la siguiente reunión.

Reply:
He resuelto esto bindeando el **this** en la llamada